src/libs/actions/PersonalDetails.js

```diff
@@ -236,6 +237,56 @@ function setPersonalDetails(details) {
     mergeLocalPersonalDetails(details);
 }
 
+/**
+ * Sets the onyx with the currency list from the network
+ * @returns {Object}
+ */
+function getCurrencyList() {
+    return API.GetCurrencyList()
+        .then((data) => {
+            const currencyListObject = JSON.parse(data.currencyList);
+            Onyx.merge(ONYXKEYS.CURRENCY_LIST, currencyListObject);
+            return currencyListObject;
+        });
+}
+
+/**
+ * Fetches the Currency preferences based on location
+ * @param {bool} withLocation
+ */
+function fetchCurrencyPreferences(withLocation) {
+    let coords = {};
+    let currency = '';
+
+    if (withLocation) {
+        Geolocation.getCurrentPosition((position) => {
+            coords = {
+                longitude: position.coords.longitude,
+                latitude: position.coords.latitude,
+            };
+        });
+        Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS,
+            {
+                isCurrencyPreferencesSaved: true,
+            });
+    }
+
+    API.GetPreferredCurrency({...coords})
+        .then((data) => {
+            currency = data.currency;
+        })
+        .then(API.GetCurrencyList)
+        .then(getCurrencyList)
+        .then((currencyList) => {
+            Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS,
+                {
+                    preferredCurrencyCode: currency,
+                    preferredCurrencySymbol: currencyList[currency].symbol,
+                });
+        })
+        .catch(error => console.debug(error));
```

Comment:
What error are we expecting here? ANything specific, or is this a catch-all?

Reply:
Maybe `.catch(error => console.debug(''Error fetching currency preference: , error));`